src/data-controller.ts

```diff
@@ -11,14 +11,18 @@ function generateProductHTML(product: Product): string {
 
 function renderProducts(prods: Product[]): void {
     const container = document.getElementById("main-container");
+    if (container) container.innerHTML = "";
     prods.map((product) => {
         if (!container) return;
```

Comment:
Merge this condition check with the above check. There is no point in checking if `container` exists on each iteration. Add the following code to line 14:

`container ? container.innterHTML = "" : return`

Reply:
Fixed. Thank you for the suggestion.